mantidproject/mantid
# Empty default facility and instrument

**Description of work.**

A new empty facility with empty instrument is the default facility.

User has to select their choice of facility for the first time, including ISIS, and instrument.

Before, if the facility in Mantid.user.properties is empty (for whatever reason), in the GUI
it was inconsistently showing it to be ISIS.

Now, if the facility and instrument are indeed empty in Mantid.user.properties file, it will be consistently 
reflected in the GUI and user will have to select their choices.

**Report to:**  Vain

**To test:**

1. Set facility value in Mantid.user.properties file to be empty. (Re)Open workbench and check 
    if it is consistently reflected in the workbench GUI (that the facility and instrument are indeed empty)

2. Select required facility and instrument, and also check 'Do not show again until next version'

3. Close and reopen workbench. First time set-up window should not appear now. 

Refs #29956 

Fixes #29956 and close #29956 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
